ai/worker/docker.go

```diff
@@ -499,20 +513,20 @@ func (m *DockerManager) watchContainer(rc *RunnerContainer, borrowCtx context.Co
 		if failures >= maxHealthCheckFailures && time.Since(startTime) > pipelineStartGracePeriod {
 			slog.Error("Container health check failed too many times", slog.String("container", rc.Name))
 			m.destroyContainer(rc, false)
-			if rc.KeepWarm {
-				slog.Info("Container was kept warm, restarting", slog.String("container", rc.Name))
-				err := m.Warm(context.Background(), rc.Pipeline, rc.ModelID, rc.OptimizationFlags)
-				if err != nil {
-					slog.Error("Error restarting warm container", slog.String("container", rc.Name), slog.String("error", err.Error()))
-				}
-			}
 			return
 		}
 
+		rc.RLock()
+		borrowCtx := rc.BorrowCtx
+		if borrowCtx == nil {
```

Comment:
could you add a comment about what situations this would be nil? It just looks a bit odd as it means `case <-borrowCtx.Done()` will never happen.